Automattic/bugomattic
# (Real, recovered) Add page transition and navigations

#### What Does This PR Add/Change?

Gahhh, sorry this PR got so big! 🙇 A lot of these page transitions/navigations changes all go together, so it was hard to break it up without creating a big rebase chain. And this change also prompted some needed modularizations to DRY things up!

But all in all, the changes are smaller than what may first appear!

First, some modularizations:
- Creating a new standardize `Banner` component for the banners (report a new issue, and start over) at the bottom of both pages.
- Creating a re-usable `ReportIssueDropdownMenu` that can be used in both the top navbar and the report a new issue banner.
- Moving the determination of what the next step in the reporting flow into a shared selector for better reuse.
- Creating a `PageNavigationProvider` and a `usePageNavigation` hook that takes care of doing all the side-effects when we change pages (like focusing the header, scrolling to the top, updating the title).

With that, we could finish everything up!
- Adding a real navbar to the top of the site
- Adding the "report an issue" banner at the bottom of the duplicate searching page
- Updating the "start over" banner at the bottom of the reporting flow page

#### Testing Instructions

- [ ] Axe devtools is clean

A quick heads-up: there is handling in there to not focus on the page subheading (like "Report a new issue") when we first render the app. However, we use `React.StrictMode` to catch errors while developing, and `StrictMode` famously renders everything twice. So, that behavior won't work quite right locally. But it will in production! If you want to see for yourself, you can just delete the `StrictMode` wrappers in `index.tsx` 🙂 .


#### Issues



Related to #  
Closes #64
